### PR TITLE
Ensure wallet without permissions can be deserialized

### LINKED ISSUE #5934: Crash of wallet service on permissions update
```
code.vegaprotocol.io/vega/wallet/wallet.(*HDWallet).UpdatePermissions(0xc000f5e080, {0xc000e47f18, 0x3}, {{{0x2f4021a, 0x4}, {0x0, 0x0, 0x0}}})
```

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #5934

### CODE REVIEW

**Review by jeremyletang — DISMISSED** [review]

**Review by candida-d — APPROVED** [review]

File: wallet/wallet/hd_wallet.go [hunk 1]
```diff
@@ -326,6 +326,11 @@ func (w *HDWallet) MarshalJSON() ([]byte, error) {
 		Keys:        w.keyRing.ListKeyPairs(),
 		Permissions: w.permissions,
 	}
+
+	if jsonW.Permissions == nil {
```
Comment (ze97286) [inline wallet/wallet/hd_wallet.go:330]: how can jsonW.Permissions be nil? it's a map not a pointer to a map isn't it?

Comment (ValentinTrinque) [reply wallet/wallet/hd_wallet.go:330]: AFAIK, well that was my mistake. I never though map could be nil.

https://go.dev/play/p/57tl546Uth3

Seems to be possible.

Comment (ze97286) [reply wallet/wallet/hd_wallet.go:330]: your playground is not what's happening in your code though and doesn't explain why you needed this change. This is more like it: https://go.dev/play/p/y6BHV9NYOvu

Comment (ze97286) [reply wallet/wallet/hd_wallet.go:330]: but to be fair `tStruct.testMap == nil` evaluates to true. so you're right.

**Review by ze97286 — APPROVED** [review]

**Pull request merged**